wordleSolver.py

```diff
@@ -0,0 +1,413 @@
+		self.coda += self.consonants
+		self.suffix = ['s']
+
+		self.green_letters = {}
+		self.yellow_letters = {} #history of all index to yellow_letters
+		self.yellow_letters_list = [] #updated list of current yellow_letters
+		self.black_letters = set()
+		self.operations = 0
+		#need a separate list to contain a "set" of the yellow letters, that allows double letters. 
+		#The yellow_letters dictionary contains a history of all guesses. 
+		#But if position is wrong twice in a row it needs to be differentiated from two of the same letter
+
+	def playGame(self, knownWord = None, dictionary = wordleDict, numLetters = 5):
+		''' 
+		knownWord is used by the simulator to test the code
+		If there is no knownWord the user must provide feebdack
+		It uses an initial guess hard coded in function
+		The user must enter feedback in a string following the following convention:
+		0 - letter not in word
+		1 - letter in word, but in wrong position
+		2 - letter in word at correct position
+		example: word is storm
+		Guess is pools
+		User should enter 00201
+		'''
+		initial_guess = 'tears'
+		print("guess is '", initial_guess, "'" )
+		if not knownWord:
+			feedback = input("feedback is: ")
+		else:
+			feedback = self.getFeedback(initial_guess, knownWord)
+		self.updateInfo(initial_guess, feedback)
+		while len(self.green_letters) != 5:
+			guess = self.guessWord(dictionary, numLetters)
+			if not guess:
+				break
+			print("guess is '", guess, "'" )
+			if not knownWord:
+				feedback = input("feedback is: ")
+			else:
+				feedback = self.getFeedback(guess, knownWord)	
+			self.updateInfo(guess, feedback)
+		guesses.append(self.operations)
+		if len(self.green_letters) != 5:
+			print ("I'm stumped")
+			return False
+		else:
+			print ("Solved it!")
+			return True
+
+	def updateInfo(self, guess, result):
+		self.updateKnownLetters(guess, result)
+		self.updatePhonemes()
+
+	def updateKnownLetters(self, guess, result):
+		''' guess is original word
+		    result is the evaluation
+		    0 is not in word
+		    1 is in word but different position
+		    2 is in word at this position
+		'''
+		#Need to consider how guessing 2 A's and only having 1 a impacts this
+		self.yellow_letters_list = [] #gets reset every time
+		for i in range(len(result)):
+			letter = guess[i]
+			r = result[i]
+			if r == '0':
+				self.black_letters.add(letter)
+			elif r == '1':
+				if i in self.yellow_letters.keys():
+					self.yellow_letters[i].append(letter)
+				else:
+					self.yellow_letters[i] = [letter]
+				self.yellow_letters_list.append(letter)
+			elif r == '2':
+				#If upgraded from yellow, need to now delete it from yellow so not double counted
+				if letter in self.yellow_letters_list:
+					index = self.yellow_letters_list.index(letter)
+					self.yellow_letters_list.pop(index)
```

Comment:
Does each letter only exist once in this list? If so we can just do this. It removes all occurrences of the letter if it exists
```suggestion
				self.yellow_letters_list.remove(letter)
```